src/coreclr/vm/precode_portable.cpp

```diff
@@ -29,6 +29,16 @@ void* PortableEntryPoint::GetActualCode(TADDR addr)
     return portableEntryPoint->_pActualCode;
 }
 
+void PortableEntryPoint::SetActualCode(TADDR addr, void* actualCode)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(actualCode != NULL);
+
+    PortableEntryPoint* portableEntryPoint = ToPortableEntryPoint(addr);
+    _ASSERTE(portableEntryPoint->_pActualCode == NULL);
```

Comment:
I do not expect this assert to hold. This is likely going to be updated in lock-free manner.